store/mongo/datastore_mongo.go

```diff
@@ -2734,3 +2740,37 @@ func (db *DataStoreMongo) UpdateDeploymentsWithArtifactName(
 func (db *DataStoreMongo) GetTenantDbs() ([]string, error) {
 	return migrate.GetTenantDbs(context.Background(), db.client, mstore.IsTenantDb(DbName))
 }
+
+func (db *DataStoreMongo) SaveLastDeviceDeploymentStatus(
+	ctx context.Context,
+	deviceDeployment model.DeviceDeployment,
+) error {
+	tenantId := ""
+	id := identity.FromContext(ctx)
+	if id != nil {
+		tenantId = id.Tenant
+	}
+	filter := bson.M{
+		"_id": deviceDeployment.DeviceId,
+	}
+
+	lastStatus := model.DeviceDeploymentLastStatus{
+		DeviceId:               deviceDeployment.DeviceId,
+		DeploymentId:           deviceDeployment.DeploymentId,
+		DeviceDeploymentId:     deviceDeployment.Id,
+		DeviceDeploymentStatus: deviceDeployment.Status,
+		TenantId:               tenantId,
+	}
+
+	database := db.client.Database(DatabaseName)
+	collDevs := database.Collection(CollectionDevicesLastStatus)
+	var err error
+	if deviceDeployment.Status.Successful() {
```

Comment:
maybe simply always replacing status in case device deployment has finished will be simpler?

Reply:
you are right.